### ISSUE TITLE
[Bug]: Disabled state not consistently applied across command bar items

### ISSUE THREAD

**Issue by sethreidnz** [microsoft/fluentui#27536]
### Library

React / v8 (@fluentui/react)

### System Info

```shell
System:
    OS: macOS 13.3
    CPU: (10) arm64 Apple M1 Max
    Memory: 392.30 MB / 32.00 GB
    Shell: 5.9 - /bin/zsh
  Browsers:
    Chrome: 112.0.5615.49
    Firefox: 105.0.2
    Safari: 16.4
```

### Are you reporting Accessibility issue?

Yes

### Reproduction

https://codepen.io/justsayno/pen/gOBPOGz

## Actual Behavior

When providing "disabled" to CommandBar items it does not consistently render disabled="true" for all of the menu item types. I'm not sure exactly what the pattern is but if you look at the pen, which I copied directly from[the fluentui 8 docs here under the title "CommandBar with split and disabled buttons"](https://developer.microsoft.com/en-us/fluentui#/controls/web/commandbar)

Notice if you inspect each button, although they are disabled in some way, they don't all have the html attribute "disabled". The split button does. but the other buttons dont.

Share button looks like this without the attribute:

![CleanShot 2023-04-13 at 12 17 07](https://user-images.githubusercontent.com/4713713/231613483-efe21967-8482-4df2-b6c9-65e255112260.png)

But then the Upload button does

![CleanShot 2023-04-13 at 12 17 35](https://user-images.githubusercontent.com/4713713/231613534-1d8a6c63-e188-4226-8fa0-0daf3d5c3df5.png)

## Expected Behavior

That if you pass disabled then it doesn't matter what type of button it should always have `disabled="true"`

### Logs

```shell
Don't have any logs as this is reproducable in the above pen.
```

### Requested priority

High

### Are you willing to submit a PR to fix?

yes

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

**Comment by smhigley** [comment]
Hi @sethreidnz! We use the `aria-disabled` attribute for disabled command bar items instead of the HTML `disabled` attribute because we maintain the ability to arrow through disabled items, and the HTML `disabled` attribute fully prevents focusability. The reason for this is we don't want the keyboard experience to change when a menuitem is conditionally disabled. This is fairly common practice for disabled items within a composite widget: https://www.w3.org/WAI/ARIA/apg/practices/keyboard-interface/#focusabilityofdisabledcontrols

In both of your screenshots the `aria-disabled` attribute is present on the menu item, and I checked your codepen and all interactive items that were marked disabled are semantically disabled:
<img width="168" alt="screenshot of accessibility tree of a menu, showing all menuitems apart from New are disabled" src="https://user-images.githubusercontent.com/3819570/231859311-17cc955b-9262-40ce-a1c0-2bee6bc202ec.png">

Hope that helps clarify the pattern!

**Comment by sethreidnz** [comment]
> Hi @sethreidnz! We use the `aria-disabled` attribute for disabled command bar items instead of the HTML `disabled` attribute because we maintain the ability to arrow through disabled items, and the HTML `disabled` attribute fully prevents focusability. The reason for this is we don't want the keyboard experience to change when a menuitem is conditionally disabled. This is fairly common practice for disabled items within a composite widget: https://www.w3.org/WAI/ARIA/apg/practices/keyboard-interface/#focusabilityofdisabledcontrols
> 
> In both of your screenshots the `aria-disabled` attribute is present on the menu item, and I checked your codepen and all interactive items that were marked disabled are semantically disabled: <img alt="screenshot of accessibility tree of a menu, showing all menuitems apart from New are disabled" width="168" src="https://user-images.githubusercontent.com/3819570/231859311-17cc955b-9262-40ce-a1c0-2bee6bc202ec.png">
> 
> Hope that helps clarify the pattern!

Hi there. Thanks for that response makes sense. Although why do some buttons have "disabled=true" and others not? As in the screenshots the behaviour is still inconsistent? 

The reason I ask is that it makes testing more convoluted because we can't simply go `isDisabled` which is fine just wondering why some of the buttons have the attribute and others dont.

**Comment by smhigley** [comment]
@sethreidnz I'm not entirely sure I understand -- in the codepen you provided, you marked every button apart from the first ("New") with the disabled component prop. In the DOM, every button aside from "New" is disabled, and additionally they are all disabled using `aria-disabled` and not the HTML `disabled` prop. I'm not really seeing any inconsistency with the behavior.

Could you clarify what the inconsistent behavior is, and which button you're seeing it on?

**Comment by sethreidnz** [comment]
Hi there. I've annotated my screenshots to make it more clear here:

![Explaination1](https://user-images.githubusercontent.com/4713713/232934552-a6b676f5-748a-498e-87ae-54fc4a538bee.png)

Probably doesn't matter but the reason i realised this was the case is because in some cases I would be able to test `isDisabled()` and get what I would have expected, in other cases not. E.g.

![Explanation2](https://user-images.githubusercontent.com/4713713/232934679-cbfebbaa-4764-41e0-a2e5-e263e7ca2c72.png)

**Comment by smhigley** [comment]
@sethreidnz in the first screenshot, that attribute is `aria-disabled` -- it's wrapping from the previous line. In the second screenshot, the button also has `aria-disabled` on it. For the test, I'd suggest using the attribute test for all of them. `.toBeDisabled()` should not be working on any of the menuitems.

**Comment by sethreidnz** [comment]
Yes I can see that. Again I am talking about how one of them has the “disabled” and one does. It. I am NOT talking about aria disabled. I am talking about how one does in fact have the regular disabled attribute. 

And yes I can test that it has aria disabled. However that doesn’t in fact tell me that the element is in fact disabled from nok screen reader users. It only tells me it’s aria disabled. 

Again look at the screenshots. They clearly show an inconsistency in that one is “disabled=true” one does not have that attribute. And again yes they both are aria disabled. I am not talking about that. Is that clear now?

**Comment by sethreidnz** [comment]
Oh my gosh. My bad! Okay cool. Still a little weird to me to have an enabled disabled button. But if that’s the way to go!

**Comment by khmakoto** [comment]
Hi @sethreidnz, the screenshot you show above has no `disabled` attribute, like @smhigley says, what you're pointing as `disabled="true"` is actually `aria-disabled="true"`, where the `aria-` part is on the first line and the `disabled="true"` wraps into the second line.

You can see this by changing the width of your dev tools console so that the line does not wrap over that part anymore as you can see below, where you can clearly see that there is no `disabled="true"` for that button:

<img width="1267" alt="image" src="https://user-images.githubusercontent.com/7798177/234358880-4abccc3c-d102-42e0-b8c8-8b0f3f67b8ef.png">

You can even go further and try to query all elements that have `disabled` set, which will return an empty DOM list in your codepen:

<img width="960" alt="EmptyList" src="https://user-images.githubusercontent.com/7798177/234359491-09f69ba6-13b8-4daa-837c-66b0620762e0.png">

I hope that clears things up, let us know you still have any doubts.
